actionpack/lib/action_dispatch/routing/mapper.rb

```diff
@@ -2528,11 +2528,15 @@ def each
 
       DEFAULT = Object.new # :nodoc:
 
-      def initialize(set) # :nodoc:
+      def initialize(set, scope = nil) # :nodoc:
         @set = set
         @draw_paths = set.draw_paths
-        @scope = Scope.new(path_names: @set.resources_path_names)
         @concerns = {}
+
+        hash = { path_names: @set.resources_path_names }
+        hash.merge!(scope) unless scope.nil?
```

Comment:
This is the bit that concerns me - all the examples I can find of `default_scope` being defined in an app are simple hashes but there may be something else in private repos. If you feel it's necessary we can extract the body of the `scope` method to a separate`build_scope` method and then use that here instead of just merging the hash.